internal/entrypoint/run.go

```diff
@@ -175,21 +180,22 @@ func Run(ctx context.Context, config *Config, exporter otlexporters.Otlexporter,
 
 			logger.WithField("number of PowerFlexClient", len(config.PowerFlexClient)).Debug("PowerFlexClient")
 
-			for key, client := range config.PowerFlexClient {
+			config.Lock.Lock()
```

Comment:
This does not seem good. You are holding the lock for too long. Ideally you want to keep the locks held for as short a period as possible, not while making REST calls. Also perhaps wrap the lock/unlock into a method that returns a copy of the config. There will always be a period where the data is updated (e.g. kubectl apply) during usage of the old config so a copy that is returned is fine. In this way the lock is held for a shorter time.

Reply:
So, essentially, this:

```
	config.Lock.Lock()
	powerflexClients := make(map[string]*entrypoint.PowerflexClient)
	for k, v := range config.PowerFlexClient {
		powerflexClients[k] = v
	}
        config.Lock.Unlock()
```
and then use `powerflexClients` for REST requests.

Reply:
That's better but not a good design to directly manage the lock for data owned by another object. Better to hide the locking into a general method that can return a copy of the current configuration. This method can then manage the lock in one place. The code will also be cleaner and not have locks spread whenever we need the current configuration.

Reply:
Updated with @alexemc example.